Classes/common/CDTIncrementalStore.h

```diff
@@ -129,4 +59,34 @@ typedef NS_ENUM(NSInteger, CDTIncrementalStoreErrors) {
     CDTISErrorNotSupported
 };
 
+/**
+ *  The databaseName is exposed in order to be able to identify the different
+ *  CDTIncrementalStore objects. @see +storesFromCoordinator:coordinator
+ */
+@property (nonatomic, strong) NSString *databaseName;
+
+/**
+ * Create a CDTReplicator object set up to replicate changes from the
+ * local datastore to a remote database.
+ *
+ *  @param remoteURL the remote server URL to which the data is replicated.
+ *  @param error     report error information
+ *
+ *  @return a CDTReplicator instance which can be used to start and
+ *  stop the replication itself, or `nil` on error.
+ */
+- (CDTReplicator *)replicatorThatPushesToURL:(NSURL *)remoteURL withError:(NSError **)error;
+
+/**
+ * Create a CDTReplicator object set up to replicate changes from a remote database to the
+ * local datastore.
+ *
+ *  @param remoteURL the remote server URL to which the data is replicated.
+ *  @param error     report error information
+ *
+ *  @return a CDTReplicator instance which can be used to start and
+ *  stop the replication itself, or `nil` on error.
+*/
+- (CDTReplicator *)replicatorThatPullsFromURL:(NSURL *)remoteURL withError:(NSError **)error;
```

Comment:
Since these replicatorThat... methods are instance methods on CDTIncrementalStore, the user will need to first get a reference to the proper CDTIncrementalStore object in order to use them.  I think that is non-obvious, so at a minimum we should give the user some guidance on how to do that, and maybe also consider extending the CDTIncrementalStore interface to make it easier.  E.g. we could offer a class method that takes a peristentStoreCoordinator and storeURL and returns a CDTIncrementalStore object.